pkg/controller/install/openshift/openshift.go

```diff
@@ -86,6 +87,168 @@ func EnsureOpenshiftIngress(c client.Client, scheme *runtime.Scheme, instance *s
 	return nil
 }
 
+// EnsureConfigMapForCrt ensures crt injection into the configmap
+func EnsureConfigMapForCrt(c client.Client, scheme *runtime.Scheme, instance *servingv1alpha1.Install) error {
+	namespace := instance.GetNamespace()
+	if routeExists, err := kindExists(c, "route", "route.openshift.io/v1", namespace); err != nil {
+		return err
+	} else if !routeExists {
+		// Not running in OpenShift
+		return nil
+	}
+
+	if err := updateConfigMapForCrt(c, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func updateConfigMapForCrt(c client.Client, namespace string) error {
+	cmName := "config-service-ca"
+	cm := &v1.ConfigMap{}
+	if err := c.Get(context.TODO(), types.NamespacedName{Name: cmName, Namespace: namespace}, cm); err != nil {
+		if errors.IsNotFound(err) {
+			// Define a new configmap
+			cm.Name = cmName
+			cm.Annotations = make(map[string]string)
+			cm.Annotations["service.alpha.openshift.io/inject-cabundle"] = "true"
+			cm.Namespace = namespace
+			err = c.Create(context.TODO(), cm)
+			if err != nil {
+				return err
+			}
+			// ConfigMap created successfully
+			return nil
+		}
+		return err
+	}
+
+	/* Check existence of service-ca.crt in the configMap
+	/ If crt exist just return or else update the configmap with annotation */
+	if _, ok := cm.Data["service-ca.crt"]; ok {
+		// Do nothing
+		return nil
+	} else {
+		cm.Annotations = make(map[string]string)
+		cm.Annotations["service.alpha.openshift.io/inject-cabundle"] = "true"
+		cm.Namespace = namespace
+		if err := c.Update(context.TODO(), cm); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// EnsureOpenshiftRegistry ensures registry for openshift
+func EnsureOpenshiftRegistry(c client.Client, scheme *runtime.Scheme, instance *servingv1alpha1.Install) error {
+	namespace := instance.GetNamespace()
+	if routeExists, err := kindExists(c, "route", "route.openshift.io/v1", namespace); err != nil {
+		return err
+	} else if !routeExists {
+		// Not running in OpenShift
+		return nil
+	}
```

Comment:
No longer necessary to check for a route. `Configure` handles that.

Reply:
Okey